.github/workflows/deploy.yml

```diff
@@ -42,91 +40,84 @@ jobs:
         with:
           ref: ${{ github.head_ref }}
           fetch-depth: 0
-
       - name: Install terraform
-        run: |
-          if ! which terraform 2>/dev/null; then
-            sh tools/install/terraform.sh
-          fi   
-
+        run: sh tools/install/terraform.sh
       - name: Check Format
         run: terraform fmt -check -recursive -diff
-
-  AdminGUI:
+  Deployment :
     runs-on: ubuntu-latest
     steps:
-      - name: Get type of VM
-        run: |
-          lscpu
-          uname -a
-          grep MemTotal /proc/meminfo
-
       - name: Checkout
         uses: actions/checkout@v3
         with:
-          ref: ${{ github.head_ref }}
-          fetch-depth: 0
-
+           ref: ${{ github.head_ref }}
+           fetch-depth: 0
+      - name: Install Dependencies
+        uses: aneoconsulting/ArmoniK.Action.Deploy/dependencies@main
+        with:
+           docker: true
+           terraform: true
+           k3s: true
+      - name: Deploy
+        uses: aneoconsulting/ArmoniK.Action.Deploy/deploy@main
+        with:
+          type: localhost
+          ext-csharp-version: ${{ steps.worker.outputs.tag }} 
       - name: Install AWS CLI
+        run: sh tools/install/aws-cli.sh
+      - name: Get Worker Tag
+        id: worker
         run: |
-          if ! which aws 2>/dev/null; then
-            sh tools/install/aws-cli.sh
-          fi   
-        
+           TAG=$(cat versions.tfvars.json | jq -r '.armonik_versions.extcsharp')
+           echo "tag=$TAG" >> $GITHUB_OUTPUT
       - name: Get Core Tag
         id: core
         run: |
-          TAG=$(grep "core" armonik-versions.txt | tail -1 | cut -d "=" -f 2)
-          echo "tag=$TAG" >> $GITHUB_OUTPUT
+            TAG=$(cat versions.tfvars.json | jq -r '.armonik_versions.core')
+            echo "tag=$TAG" >> $GITHUB_OUTPUT
+      - name: Save artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: deployed-files
+          path: /path/to/deployed/files
 
-      - name: Get Admin GUI Tag
-        id: admin-gui
-        run: |
-          TAG=$(grep "admin-gui" armonik-versions.txt | tail -1 | cut -d "=" -f 2)
-          echo "tag=$TAG" >> $GITHUB_OUTPUT
 
-      - uses: ./
-        name: Deploy infrastructure
-        id: deploy-infra
+  AdminGUI:
+    runs-on: ubuntu-latest
+    needs: Deployment
+    steps:
+      - name: Download artifacts
+        uses: actions/download-artifact@v2
```

Comment:
I believe this will not work as the runner that deployed the infrastructure has ended.